RecoVertex/V0Producer/src/V0Fitter.h

```diff
@@ -47,18 +41,28 @@
 #include "Geometry/CommonDetUnit/interface/GeomDet.h"
 #include "Geometry/TrackerGeometryBuilder/interface/GluedGeomDet.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
-
 #include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
 #include "DataFormats/BeamSpot/interface/BeamSpot.h"
-
-#include <string>
-#include <fstream>
+#include "CommonTools/CandUtils/interface/AddFourMomenta.h"
```

Comment:
sorry this is new to me - is the general idea to have an include statement
in the header file (i.e. class declaration file) only if it is necessary in
the declaration of the class?

In my particular example, the class declaration involves a function with a
parameter of  reco::VertexCompositeCandidateCollection - in this case I
would want to have the particular include file in the same file as the
class declaration?

On Tue, Oct 6, 2015 at 2:45 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoVertex/V0Producer/src/V0Fitter.h
> <https://github.com/cms-sw/cmssw/pull/11660#discussion_r41305162>:
>
> >  #include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
> >  #include "DataFormats/BeamSpot/interface/BeamSpot.h"
> > -
> > -#include <string>
> > -#include <fstream>
> > +#include "CommonTools/CandUtils/interface/AddFourMomenta.h"
>
> better have all the includes in the .cc file: based on the changes in this
> file, these extra includes are not needed at the class declaration step.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/11660/files#r41305162>.
>